chalicelib/checks/wrangler_checks.py

```diff
@@ -710,3 +712,54 @@ def patch_wfr_and_log(wfr, full_output):
             check.summary = 'No WFR patches run'
 
     return check
+
+
+@check_function()
+def validate_entrez_geneids(connection, **kwargs):
+    ''' query ncbi to see if geneids are valid
+    '''
+    check = init_check_res(connection, 'validate_entrez_geneids')
+    problems = {}
+    timeouts = 0
+    search_query = 'search/?type=Gene&limit=all'
+    genes = ff_utils.search_metadata(search_query, key=connection.ff_keys, ff_env=connection.ff_env)
+    if not genes:
+        check.status = "FAIL"
+        check.description = "Could not retrieve gene records from fourfront"
+        return check
+    geneids = [g.get('geneid') for g in genes]
+
+    query = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=gene&id={id}"
+    for gid in geneids:
+        if timeouts > 5:
+            check.status = "FAIL"
+            check.description = "Too many ncbi timeouts. Maybe they're down."
+            return check
+        gquery = query.format(id=gid)
+        # make 3 attempts to query gene at ncbi
+        for count in range(3):
+            resp = requests.get(gquery)
+            if resp.status_code == 200:
+                break
+            if resp.status_code == 429:
+                time.sleep(0.334)
+                continue
+            if count == 2:
+                timeouts += 1
+                problems[gid] = 'ncbi timeout'
+        try:
+            rtxt = resp.text
+        except AttributeError:
+            problems[gid] = 'empty response'
+        else:
+            if rtxt.startswith('Error'):
+                problems[gid] = 'not a valid geneid'
+    if problems:
+        check.brief_output = "{} problematic entrez gene ids.".format(len(problems))
```

Comment:
This might be better as a check.summary than check.brief_output, but it's up to you.